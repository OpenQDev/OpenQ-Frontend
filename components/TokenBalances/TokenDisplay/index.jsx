import React from 'react';
import Image from 'next/image';

const TokenDisplay = ({ token, onSelect, showCursor }) => {
  return (
    <button
      aria-label={`select ${token.name || token.symbol}`}
      className={`flex flex-row ${showCursor && 'cursor-pointer'} space-x-4 pb-3`}
      onClick={() => onSelect(token)}
    >
      <div className='pt-2 h-8 w-6'>
        <Image
          src={token.logoURI || token.path || '/crypto-logos/ERC20.svg'}
          className='rounded-full'
          alt='n/a'
          width={24}
          height={24}
        />
      </div>
      <div className='flex flex-col'>
        <div className='font-bold truncate'>{token.symbol}</div>
        <div className='text-sm text-gray-200 truncate w-32'>{token.name}</div>
      </div>
    </button>
  );
};

export default TokenDisplay;
