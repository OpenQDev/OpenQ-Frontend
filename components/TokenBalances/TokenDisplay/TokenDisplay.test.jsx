import React from 'react';
import TokenDisplay from '.';
import renderer from 'react-test-renderer';
import Constants from '../../../test-utils/constant';

const token = Constants.tokenMetadata;

describe('TokenDisplay', () => {
  it('should match DOM Snapshot', () => {
    const tree = renderer.create(<TokenDisplay token={token} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
