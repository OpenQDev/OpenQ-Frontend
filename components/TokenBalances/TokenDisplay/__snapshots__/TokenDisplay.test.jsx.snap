// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TokenDisplay > should match DOM Snapshot 1`] = `
<button
  aria-label="select DAI"
  className="flex flex-row undefined space-x-4 pb-3"
  onClick={[Function]}
>
  <div
    className="pt-2 h-8 w-6"
  >
    <img
      alt="n/a"
      className="rounded-full"
      data-nimg="1"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=https%3A%2F%2Ftokens.1inch.exchange%2F0x6b175474e89094c44da98b954eedeac495271d0f.png&w=48&q=75"
      srcSet="/_next/image?url=https%3A%2F%2Ftokens.1inch.exchange%2F0x6b175474e89094c44da98b954eedeac495271d0f.png&w=32&q=75 1x, /_next/image?url=https%3A%2F%2Ftokens.1inch.exchange%2F0x6b175474e89094c44da98b954eedeac495271d0f.png&w=48&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
  </div>
  <div
    className="flex flex-col"
  >
    <div
      className="font-bold truncate"
    >
      DAI
    </div>
    <div
      className="text-sm text-gray-200 truncate w-32"
    />
  </div>
</button>
`;
